src/Security/Authentication/Certificate/src/CertificateAuthenticationHandler.cs

```diff
@@ -212,6 +212,8 @@ private X509ChainPolicy BuildChainPolicy(X509Certificate2 certificate)
                 chainPolicy.TrustMode = Options.ChainTrustValidationMode;
             }
 
+            chainPolicy.ExtraStore.AddRange(Options.AdditionalChainCertificates);
```

Comment:
Does this happen per request?

Reply:
Depends, validation only happens once per certificate if the cert cache is enabled (memory cache with 2 minute sliding expiration)

Reply:
But AdditionalChainCertificates is going to be empty almost always